helper/krylovHelper.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <iomanip>
+
+#include "krylovTimeEvolver.h"
+
+#ifdef USE_HDF
+#include <H5Cpp.h>
+#include "krylovHelper.h"
+using namespace H5;
+#endif
+
+
+//parent class
+class parameter
+{
+public:
+	parameter(const std::string& name) : p_name(name) {}
+	virtual ~parameter() {}
+	virtual std::string getData() = 0;
+	virtual int getBasicDataType() = 0;
+	virtual double retDouble() = 0;
+	virtual int retInt() = 0;
+	std::string getName();
+
+private:
+	std::string p_name;
+};
+
+//parameter class for different data types
+template <typename T>
+class typedParameter : public parameter
+{
+public:
+	typedParameter(const std::string& name, const T& data) : parameter(name), para_data(data) {}
+
+	std::string getData() //write to string
+	{
+		std::ostringstream oss;
+		oss << std::setprecision(10) << para_data; //set amount of digits in the output string
+		return oss.str();
+	}
+
+	int getBasicDataType() //Help to decide which datatype is stored in current parameter
+	{
+		if (std::is_same<T, double>::value)
+			return 1;
+		else if (std::is_same<T, int>::value)
+			return 2;
+		else if (std::is_same<T, bool>::value)
+			return 3;
+		else
+			return -1;
+	}
+
+	double retDouble() 
```

Comment:
The problem is to decide which return datatype should retValue have since it needs to be fixed. The only other option I see (with my limit knowledge) is to use void* (typeless points), which needs to be cast into to the datatype by the caller.

Reply:
I tried to be super generic here, but it is probably note required. One can also drop the generic template and just require parameters to be either double, int, or bool (which could also be encoded in int).

Reply:
I think there's a very nice solution: Use the fact that typedParameter is a class template and define

T getValue() 
	{
		return para_data;
	}

Reply:
Unfortunately, that doesn't work. paraIter is an iterator/pointer of the parent class parameter, which doesn't know about the template T. The problem is that I don't know how to combine different types of parameter in a common list/vector without the help of an abstract parent class.